operators/pkg/controller/elasticsearch/pvc/pvc.go

```diff
@@ -107,11 +106,11 @@ func compareResources(claim, candidate *corev1.PersistentVolumeClaim) bool {
 }
 
 func compareStorageClass(claim, candidate *corev1.PersistentVolumeClaim) bool {
-	if claim.Spec.StorageClassName != nil {
-		return reflect.DeepEqual(claim.Spec.StorageClassName, candidate.Spec.StorageClassName)
+	if claim.Spec.StorageClassName == nil {
+		// volumeClaimTemplate has no storageClass set
+		return true
```

Comment:
not sure I understand why we return true here? I read it as "if there is no storage class requested, then the candidate PV is compatible" which sounds wrong?

Reply:
Hmm I guess we have no way of knowing what is the "default" storage class on a given cluster, so we fallback to anything?

Reply:
Yes, that's it

Reply:
Maybe we should get that comment a bit more explicit:

```
// volumeClaimTemplate has no storageClass set: it should use the k8s cluster default
// since we don't know that default, we fallback to reusing any available volume 
// from the same cluster (whatever the storage class actually is)
```